src/NuGet.Core/NuGet.ProjectModel/PackageSpecWriter.cs

```diff
@@ -65,6 +65,12 @@ internal static void Write(PackageSpec packageSpec, IObjectWriter writer, bool h
             SetDictionaryValues(writer, "scripts", packageSpec.Scripts);
 #pragma warning restore CS0612 // Type or member is obsolete
 
+            SetValue(writer, "SdkAnalysisLevel", packageSpec.SdkAnalysisLevel);
+
+            if (packageSpec.UsingMicrosoftNETSdk is not null)
+            {
+                SetValue(writer, "UsingMicrosoftNETSdk", packageSpec.UsingMicrosoftNETSdk.ToString());
```

Comment:
JSON supports bools, so write it as a bool rather than a string.

Also, given we default to false, we can write the value only when the value is true, so that parsing (reading) has a very minor perf benefit when reading the default value.

Although considering SDK style projects are more common and the only .NET project types that are "moving forward", perhaps we should default to true in the reader (meaning you need to handle the case when the property isn't defined in the json), and therefore only write to JSON when the value is false.

Reply:
That is a good idea for performance. However, currently we default to null, unless we clearly get the information that UsingMicrosoftNetSdk is true or false.  Which i think is good for correctness.

Reply:
Null doesn't have a semantic meaning. It means the same thing as false. So, I don't understand what you mean by correctness. When we use this property to make decisions about what warnings and errors to enable, null will have the same code path as false, always.

You could make it non-nullable in `PackageSpec.cs` and define the default value `public bool UsingMicrosoftNETSdk { get; set; } = true;`, so it will only be set to false if something explicit sets it to false, which we'd need to do everywhere we're doing `bool.TryParse` (if try returns false, set the value to false).

Reply:
Defaulting to false when the `UsingMicrosoftNETSdk` value is absent might imply that we are certain about its value, even when it's actually undefined. I was wondering if there might be cases where the value isn't written to the dgSpec file. In such instances, should we automatically assume it's false instead of null, which typically indicates an unknown state?